.github/workflows/lint.yml

```diff
@@ -18,31 +18,38 @@ jobs:
     name: eslint
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
     - name: Use desired version of NodeJS
-      uses: actions/setup-node@v4
+      uses: actions/setup-node@cdca7365b2dadb8aad0a33bc7601856ffabcc48e # v4.3.0
       with:
           node-version-file: '.nvmrc'
+
     - name: Cache Node
-      uses: actions/cache@v4
+      uses: actions/cache@d4323d4df104b026a6aa633fdb11d772146be0bf # v4.2.2
       with:
         path: |
           node_modules
           ~/.cache
           ~/.npm
         key: ${{ hashFiles('package-lock.json') }}
+
     - name: Check versions
       run: npm -v; node -v
+
     - name: npm install
       run: npm install
+
     - name: Generate linting report
       run: npm run lint:js -- --output-file eslint-report.json --format json
       continue-on-error: true
+
     - name: Annotate code linting results
-      uses: ataylorme/eslint-annotate-action@1.2.0
+      uses: ataylorme/eslint-annotate-action@5f4dc2e3af8d3c21b727edb597e5503510b1dc9c # v2.2.0
```

Comment:
Looks like some interesting updates in the 3.0.0 release, but definitely a concern for a different day/PR

Reply:
Yeah... I debated going to 3.0.0 but with the noted breaking changes there, decided it might be better to handle that separately (though it may just work without any needed changes on our end)